### PR TITLE
[WIP] C++ 17 upgrade

### PR DESCRIPTION
**Pull request by rete**
BEGINRELEASENOTES
- Switched ON C++ 17 flag
- Pass correctly cxx/c standard to following packages:
   - Boost (b2 args)
   - GSL: Passing c11 standard (latest for C)
   - Fastjet: via configure command (using CXXFLAGS)
- Fastjet: set flag `--enable-auto-ptr=no` to be compatible with c++17
- Added function to add a CMake cache variable per module using `addCMakeCache()`
- Corrected all paths to CMake config files to be exported in ILCSoft.cmake

ENDRELEASENOTES

Note:
- `CondDBMySQL` needed an upgrade. See: https://github.com/iLCSoft/CondDBMySQL/pull/3 
- `KiTrackMarlin` needed an upgrade. See: https://github.com/iLCSoft/KiTrackMarlin/pull/5
- `ForwardTracking` needed an upgrade. See: https://github.com/iLCSoft/ForwardTracking/pull/10

TODO: 

- [x] Upgrade CondDBMySQL version. See https://github.com/iLCSoft/CondDBMySQL/pull/3
- [ ] Need a C++ 17 MarlinTPC version

Fixes https://github.com/iLCSoft/iLCInstall/issues/76

### CODE REVIEW

**Comment by rete** [comment]
Full installation working in the following conditions:
- Using CondDBMySQL HEAD version (with c++ 17 fixes) instead of latest
- Removed MarlinTPC package (not compatible with c++17)
- + Changes in this pull request

Tested in docker ubuntu 1.04, GCC 8.3

File: ilcsoft/mysql.py [hunk 1]
```diff
@@ -46,4 +46,5 @@ def postCheckDeps(self):
             self.envpath["LD_LIBRARY_PATH"].append( "$MYSQL_HOME/lib" )
         self.envpath["LD_LIBRARY_PATH"].append( "$MYSQL_HOME/lib64/mysql" )
         self.envpath["LD_LIBRARY_PATH"].append( "$MYSQL_HOME/lib/mysql" )
+        self.addCMakeCache( "MySQL_DIR", self.installPath, "Path to MySQL directory" )
```
Comment (rete) [inline ilcsoft/mysql.py]: @petricm This should fix the issue in https://github.com/iLCSoft/CondDBMySQL/pull/3/files#diff-ab792ce188708a5be46575410c93afeaR9 . After this PR is merged I'll update the TravisCI of CondDBMySQL.

**Comment by gaede** [comment]
On SL6 I get the following problem w/ Fastjet:
```
Making all in src
make[1]: Entering directory `/afs/desy.de/project/ilcsoft/sw/x86_64_gcc82_sl6/FastJet/3.2.1/build/src'
/bin/sh ../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../FastJet/src -I../include/fastjet    -g  -Woverloaded-virtual  -DDROP_CGAL  -I../../FastJet/src/../include CXXFLAGS -MT libfastjet_la-DnnPlane.lo -MD -MP -MF .deps/libfastjet_la-DnnPlane.Tpo -c -o libfastjet_la-DnnPlane.lo `test -f 'DnnPlane.cc' || echo '../../FastJet/src/'`DnnPlane.cc
/bin/sh ../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../FastJet/src -I../include/fastjet    -g  -Woverloaded-virtual  -DDROP_CGAL  -I../../FastJet/src/../include CXXFLAGS -MT libfastjet_la-Dnn4piCylinder.lo -MD -MP -MF .deps/libfastjet_la-Dnn4piCylinder.Tpo -c -o libfastjet_la-Dnn4piCylinder.lo `test -f 'Dnn4piCylinder.cc' || echo '../../FastJet/src/'`Dnn4piCylinder.cc
/bin/sh ../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../FastJet/src -I../include/fastjet    -g  -Woverloaded-virtual  -DDROP_CGAL  -I../../FastJet/src/../include CXXFLAGS -MT libfastjet_la-Dnn3piCylinder.lo -MD -MP -MF .deps/libfastjet_la-Dnn3piCylinder.Tpo -c -o libfastjet_la-Dnn3piCylinder.lo `test -f 'Dnn3piCylinder.cc' || echo '../../FastJet/src/'`Dnn3piCylinder.cc
/bin/sh ../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../FastJet/src -I../include/fastjet    -g  -Woverloaded-virtual  -DDROP_CGAL  -I../../FastJet/src/../include CXXFLAGS -MT libfastjet_la-Dnn2piCylinder.lo -MD -MP -MF .deps/libfastjet_la-Dnn2piCylinder.Tpo -c -o libfastjet_la-Dnn2piCylinder.lo `test -f 'Dnn2piCylinder.cc' || echo '../../FastJet/src/'`Dnn2piCylinder.cc
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../FastJet/src -I../include/fastjet -g -Woverloaded-virtual -DDROP_CGAL -I../../FastJet/src/../include CXXFLAGS -MT libfastjet_la-DnnPlane.lo -MD -MP -MF .deps/libfastjet_la-DnnPlane.Tpo -c ../../FastJet/src/DnnPlane.cc 
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../FastJet/src -I../include/fastjet -g -Woverloaded-virtual -DDROP_CGAL -I../../FastJet/src/../include CXXFLAGS -MT libfastjet_la-Dnn2piCylinder.lo -MD -MP -MF .deps/libfastjet_la-Dnn2piCylinder.Tpo -c ../../FastJet/src/Dnn2piCylinder.cc 
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../FastJet/src -I../include/fastjet -g -Woverloaded-virtual -DDROP_CGAL -I../../FastJet/src/../include CXXFLAGS -MT libfastjet_la-Dnn3piCylinder.lo -MD -MP -MF .deps/libfastjet_la-Dnn3piCylinder.Tpo -c ../../FastJet/src/Dnn3piCylinder.cc 
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../FastJet/src -I../include/fastjet -g -Woverloaded-virtual -DDROP_CGAL -I../../FastJet/src/../include CXXFLAGS -MT libfastjet_la-Dnn4piCylinder.lo -MD -MP -MF .deps/libfastjet_la-Dnn4piCylinder.Tpo -c ../../FastJet/src/Dnn4piCylinder.cc 
g++: error: CXXFLAGS: No such file or directory
g++: error: CXXFLAGS: No such file or directory
g++: error: CXXFLAGS: No such file or directory
g++: error: CXXFLAGS: No such file or directory
```

**Comment by rete** [comment]
@gaede Don't forget to push the fixes (git and FastJet CXXFLAGS)

**Comment by gaede** [comment]
done.

**Pull request merged**